File: src/SSHServer/SSHServerException.cs
using SSHServer.Packets;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer
{
    public class SSHServerException : Exception
    {
        public DisconnectReason Reason { get; set; }

        public SSHServerException(DisconnectReason reason, string message) : base(message)
        {
            Reason = reason;
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/SSHServerException.ts
import { DisconnectReason } from "./Packets/DisconnectReason";

export class SSHServerException extends Error {
    public reason: DisconnectReason;

    constructor(reason: DisconnectReason, message: string) {
        super(message);

        this.reason = reason;
    }
}

export * from "./Packets/DisconnectReason";
